test/functional/api/v2/template_controller_test.rb

```diff
@@ -24,6 +25,146 @@ class TemplateControllerTest < ::ActionController::TestCase
           refute Dir.entries("#{tmpdir}/provisioning_templates/provision").include?('export_test_template.erb')
         end
       end
+
+      test "should import filtered templates" do
+        post :import, params: { "repo" => "#{ForemanTemplates::Engine.root}/test/templates/core", "filter" => 'filter1', "prefix" => "filtTemp_" }
+        assert_response :success
+        templates = JSON.parse(@response.body)['message']['templates']
+        templates.each do |template|
+          if template['name'] == 'filtTemp_filter1 template'
```

Comment:
`filtTemp_filter1 template` was not imported due to missing template kind.

Reply:
But, It is there mentioned already in filter1 template -> https://github.com/theforeman/foreman_templates/pull/104/files#diff-f78c60bb7753401d1ec4a66c7b2630d3R2. It should be a bug.

Reply:
Not really. You need to specify you want to create associations based on [metadata](https://github.com/theforeman/foreman/blob/develop/app/models/template.rb#L215)

```ruby
post :import, params: { "repo" => "#{ForemanTemplates::Engine.root}/test/templates/core", 
                                    "filter" => 'filter1',
                                    "prefix" => "filtTemp_",
                                     "associate" => "new" }
```